src/LanguageServer/Impl/Implementation/Server.OnTypeFormatting.cs

```diff
@@ -49,7 +50,15 @@ public sealed partial class Server {
                 }
 
                 var lineFormatter = new LineFormatter(reader, Analyzer.LanguageVersion);
-                return lineFormatter.FormatLine(targetLine);
+                var edits = lineFormatter.FormatLine(targetLine);
+                var unmatchedToken = lineFormatter.UnmatchedToken(targetLine);
+
+                if (unmatchedToken != null) {
+                    var message = Resources.LineFormatter_UnmatchedToken.FormatInvariant(unmatchedToken.Value.token, unmatchedToken.Value.line + 1);
+                    ShowMessage(MessageType.Warning, message);
```

Comment:
This will pop up message box (modal in VS) that has to be dismissed. UI designers consider popups intrusive and the practice is discouraged. Non-intrusive console output (via logger) is  typically sufficient.

Reply:
I switched it out for `LogMessage`.